ui/app/src/components/Search/utils.ts

```diff
@@ -279,7 +279,7 @@ export const useSearchState = ({ searchParameters, onSelect }: useSearchStatePro
   const refreshSearch = useCallback(() => {
     setError(null);
     setIsFetchingResults(true);
-    search(site, searchParameters).subscribe({
+    search(site, { ...searchParameters, keywords: encodeURIComponent(searchParameters.keywords) }).subscribe({
```

Comment:
Encode everything at the service. The site id should be encoded too, use the QS util.